### ISSUE TITLE
Styles are not combined - webpack 4

### ISSUE THREAD

**Issue by qodesmith** [ChristianMurphy/postcss-combine-duplicated-selectors#219]
I'm not seeing styles combined for the same selectors when using Webpack 4. Hoping I'm missing something simple here.

**postcss.config.js**
```javascript
module.exports = {
  plugins: {
    'autoprefixer': {},
    'css-mqpacker': { sort: true }, // https://goo.gl/wJoJtV
    'postcss-discard-comments': { removeAll: true }, // https://goo.gl/a4ubai
    'postcss-combine-duplicated-selectors': {
      removeDuplicatedProperties: true
    }
  }
}
```

**Relevant webpack portion:**
```javascript
module: {
  rules: [
    {
      test: /\.(scss|css)$/,
      include: path.resolve(__dirname, 'src'),
      use: [
        env.prod ? MiniCssExtractPlugin.loader : 'style-loader', // https://goo.gl/uUBr8G
        {
          loader: 'css-loader',
          options: {
            importLoaders: 2
          }
        },
        'postcss-loader', // https://goo.gl/BCwCzg - needs to be *after* `css-loader`.
        {
          loader: 'sass-loader',
          options: {
            includePaths: [
              'node_modules/sassyons'
            ]
          }
        }
      ]
    }
  ]
}
```

**CSS output:**
```css
*{
  box-sizing:border-box; }

html,
body{
  margin:0;
  padding:0;
  width:100%;
  height:100%;
  font:24px/1 Arial, Helvetica, sans-serif; }

.bg-gold{
  background-color:#ffd700; }

.i{
  font-style:italic; }

.fw4{
  font-weight:400; }

 .home-ac{
  height:100%; }

 .home-ac{
  position:fixed; }

.bg-black-80{
  background-color:rgba(0, 0, 0, 0.8); }

.white-80{
  color:rgba(255, 255, 255, 0.8); }

 .home-ac{
  width:100%; }
```

The `.home-ac` class is declared 3 times and not combined. Have I set things up wrong?

**Comment by qodesmith** [comment]
And to make sure that the build process was hitting the `postcss.config.js` file, I renamed `postcss-combine-duplicated-selectors` to `postcss-combine-duplicated-selectorz` and it threw an error. So at least we know it's reading the config file.

**Comment by ChristianMurphy** [comment]
I added this sample code as a test case in #220.
It looks like it is de-duplicating as expected when parsed as css. :eyes: 

I'd try disabling the other postcss plugins and see if that resolves the issue. :thinking:

**Comment by qodesmith** [comment]
@ChristianMurphy Thanks for such a quick reply. So I commented out all the other plugins in my `postcss.config.js` file, but there is still duplicates. It's the same output as above 😞 Do you think any of this has something to do with using `MiniCssExtractPlugin` in my webpack file? The webpack docs say to stay away from `ExtractTextPlugin`. My guess is the issue isn't in postcss land, rather, in webpack land.

**Comment by ChristianMurphy** [comment]
It's possible, 🤔 
does the issue persist when MiniCssExtract loader is commented out and `style-loader` is applied instead?

**Comment by qodesmith** [comment]
Doing so produces a single JS asset with all the CSS in there, it doesn't generate a separate CSS file.

**Comment by ChristianMurphy** [comment]
Are any of the other postcss plugins taking affect when MiniCssExtract loader is enabled?
E.G. `postcss-discard-comments` removing

```css
/* test comment, should go away */
```

**Comment by qodesmith** [comment]
I found the issue! Ok, let's see if I can spit this out right...

**TL;DR** - Webpack plugins run after loaders so deduping selectors was happening *before* a plugin was able to manipulate the assets.

I'm using the phenomenal [purgecss-webpack-plugin](https://github.com/FullHuman/purgecss-webpack-plugin) to remove unused selectors from my final asset. `postcss-combine-duplicated-selectors` was running (via the postcss loader) prior to purgecss and was getting presented with something like this:

```css
.h-100, .home-ac {
  height: 100%;
}

.fixed, .home-ac {
  position: fixed;
}
```

I'm assuming at this point since these are 2 different selectors there was nothing to combine. The purgecss plugin would then strip out the `.h-100` and `.fixed` classes, leaving multiple instances of `.home-ac`. Soooo.... this was an interesting lesson in the order of operations in webpack land.

The good news is that there is a [postcss-purgecss](https://github.com/FullHuman/postcss-purgecss) plugin that let's me handle this in the loader stage in webpack. I tried it and it works! Thanks for all your help and I hope that anyone googling find this helpful.
